ryan_savill/principal_feature_analysis/readme.md

```diff
@@ -0,0 +1,902 @@
+    <tr>
+      <th>minor_axis_length</th>
+      <td>0.308058</td>
+      <td>0.048526</td>
+      <td>0.032858</td>
+      <td>-0.149205</td>
+      <td>-0.075049</td>
+    </tr>
+    <tr>
+      <th>solidity</th>
+      <td>0.139750</td>
+      <td>0.213562</td>
+      <td>0.179860</td>
+      <td>-0.784037</td>
+      <td>0.109756</td>
+    </tr>
+    <tr>
+      <th>mean_intensity</th>
+      <td>-0.298859</td>
+      <td>-0.031721</td>
+      <td>-0.044582</td>
+      <td>-0.053140</td>
+      <td>-0.290490</td>
+    </tr>
+    <tr>
+      <th>max_intensity</th>
+      <td>-0.223891</td>
+      <td>0.445127</td>
+      <td>-0.015072</td>
+      <td>-0.078562</td>
+      <td>-0.095429</td>
+    </tr>
+    <tr>
+      <th>min_intensity</th>
+      <td>-0.248677</td>
+      <td>-0.277709</td>
+      <td>-0.003527</td>
+      <td>-0.151868</td>
+      <td>-0.443942</td>
+    </tr>
+    <tr>
+      <th>image_stdev</th>
+      <td>-0.124373</td>
+      <td>0.597732</td>
+      <td>-0.026935</td>
+      <td>0.059831</td>
+      <td>0.275986</td>
+    </tr>
+    <tr>
+      <th>avg distance of 2 closest points</th>
+      <td>0.312403</td>
+      <td>0.040115</td>
+      <td>0.025366</td>
+      <td>-0.113510</td>
+      <td>-0.083929</td>
+    </tr>
+    <tr>
+      <th>avg distance of 3 closest points</th>
+      <td>0.315714</td>
+      <td>0.031638</td>
+      <td>0.028109</td>
+      <td>-0.084834</td>
+      <td>-0.084732</td>
+    </tr>
+    <tr>
+      <th>avg distance of 4 closest points</th>
+      <td>0.316944</td>
+      <td>0.023361</td>
+      <td>0.033518</td>
+      <td>-0.059729</td>
+      <td>-0.078356</td>
+    </tr>
+    <tr>
+      <th>touching neighbor count</th>
+      <td>-0.190434</td>
+      <td>-0.078438</td>
+      <td>-0.284396</td>
+      <td>-0.437245</td>
+      <td>-0.339335</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+If you have read the paper or are following along with the paper we have just completed step 3 of the explained algorithm. What is done now is that we perform clustering on the matrix A_q with K-means clustering. Essentially what we are trying to do is find clusters of features as features that are clustered in this matrix have similar influences on the principal components and are thus correlating and not telling us much. This way we can just choose the feature that is closest to the cluster centre and it will be chosen as a principal feature. 
+
+We can see that the choice of the cluster-number will determine how many features we choose. Y. Lu et al. recommend choosing the cluster-number: p, as: p >= q because we can't be completely sure that the variance that we chose is explained when p = q. We can just opt for a slightly larger p (arbitrary choice of 2  larger than q) but there is room for improvement in terms of an automated choice here OR it could be a parameter that we can implement later.
+
+Since K-means is implemented well in scikit-learn this procedure is just a couple of lines:
```

Comment:
```suggestion
Since K-means-clustering is implemented well in [scikit-learn](https://scikit-learn.org/stable/modules/clustering.html) this procedure is just a couple of lines:
```